src/MiddlewareListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace DI\Bridge\Silex;
+
+use DI\Bridge\Silex\Application;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Replacement for the Silex MiddlewareListener to allow arbitrary injection into middleware functions.
+ *
+ * @author Felix Becker <f.becker@outlook.com>
+ */
+class MiddlewareListener extends \Silex\EventListener\MiddlewareListener
```

Comment:
I'm not sure I understand when this class is used by Silex? Since `App::before()` and `App::after()` are overridden, why do we need to override this too?